Framework/PythonInterface/plugins/algorithms/PelicanCrystalProcessing.py

```diff
@@ -0,0 +1,219 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import os
+import h5py
+
+from mantid import config
+from mantid.kernel import (Direction, StringArrayMandatoryValidator, StringArrayProperty, CompositeValidator)
+from mantid.api import (PythonAlgorithm, FileProperty, FileAction, Progress)
+from mantid.simpleapi import *  # noqa
+
+
+def seq_to_list(iseqn):
```

Comment:
Could we add some Python typing (https://docs.python.org/3.8/library/typing.html) to the free functions here? It will help with readability of the code. I intend to push for making it mandatory soon so it will be good to get in the habit. It would also be good to add a brief documentation string to each to signal its intent.

Reply:
I've added python typing to the functions based on the 3.8 standard but it's probably more natural from 3.9 onwards when std python types can be used as generic types.